cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -100,6 +105,32 @@ private void checkServiceResources(Set<String> problemKeys) {
     }
   }
 
+  private void checkBindAddresses() {
+    // check if service bind addresses are loopback addresses
+    ImmutableSet.Builder<String> builder = ImmutableSet.<String>builder()
+      .add(Constants.AppFabric.SERVER_ADDRESS)
+      .add(Constants.Dataset.Executor.ADDRESS)
+      .add(Constants.Stream.ADDRESS)
+      .add(Constants.Router.ADDRESS)
+      .add(Constants.Metrics.ADDRESS)
+      .add(Constants.LogSaver.ADDRESS)
+      .add(Constants.Explore.SERVER_ADDRESS)
+      .add(Constants.Metadata.SERVICE_BIND_ADDRESS);
+    Set<String> bindAddressKeys = builder.build();
+
+    for (String bindAddressKey : bindAddressKeys) {
+      String bindAddress = cConf.get(bindAddressKey);
+      try {
+        if (InetAddress.getByName(bindAddress).isLoopbackAddress()) {
+          LOG.warn("{} is set to {}. The service may not be discoverable on a multinode Hadoop cluster.",
```

Comment:
Shouldn't this error out, why show warning and not stop from proceeding further. It's going to be useless anyways in most cases.